<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Essential parameters -->
  <arg name="map_path" default="$(env HOME)/autoware_map/sample-map-planning" description="point cloud and lanelet2 map directory path"/>
  <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
  <arg name="sensor_model" default="sample_sensor_kit" description="sensor model name"/>
  <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>

  <!-- launch module preset -->
  <arg name="planning_module_preset" default="default" description="planning module preset"/>

  <!-- Optional parameters -->
  <!-- Map -->
  <arg name="lanelet2_map_file" default="lanelet2_map.osm" description="lanelet2 map file name"/>
  <arg name="pointcloud_map_file" default="pointcloud_map.pcd" description="pointcloud map file name"/>
  <!-- System -->
  <arg name="launch_system_monitor" default="false" description="launch system monitor"/>
  <arg name="launch_dummy_diag_publisher" default="false" description="launch dummy diag publisher"/>
  <!-- Scenario simulation -->
  <arg name="initial_engage_state" default="true" description="/vehicle/engage state after starting Autoware"/>
  <arg name="perception/enable_detection_failure" default="true" description="enable to simulate detection failure when using dummy perception"/>
  <arg name="perception/enable_object_recognition" default="true" description="enable object recognition when using dummy perception"/>
  <arg name="perception/use_base_link_z" default="true" description="dummy perception uses base_link z axis coordinate if it is true"/>
  <arg name="sensing/visible_range" default="300.0" description="visible range when using dummy perception"/>
  <arg name="scenario_simulation" default="false" description="use scenario simulation"/>
  <!-- Tools -->
  <arg name="rviz" default="true" description="launch rviz"/>
  <arg name="rviz_config" default="$(find-pkg-share load_vehicle)/rviz/autoware.rviz" description="rviz config"/>
  <let name="rviz_respawn" value="false" if="$(var scenario_simulation)"/>
  <let name="rviz_respawn" value="true" unless="$(var scenario_simulation)"/>
  <!-- Vcu emulation -->
  <arg name="vehicle_simulation" default="true" description="use vehicle simulation"/>
  
  <node pkg="load_vehicle" exec="param_pub" name="pub" namespace="param_pub1">
    <param name="map_path" value="$(var map_path)"/>
    <param name="vehicle_model" value="$(var vehicle_model)"/>
    <param name="sensor_model" value="$(var sensor_model)"/>
    <param name="data_path" value="$(var data_path)"/>
    <param name="planning_module_preset" value="$(var planning_module_preset)"/>
    <param name="lanelet2_map_file" value="$(var lanelet2_map_file)"/>
    <param name="pointcloud_map_file" value="$(var pointcloud_map_file)"/>
    <param name="launch_system_monitor" value="$(var launch_system_monitor)"/>
    <param name="launch_dummy_diag_publisher" value="$(var launch_dummy_diag_publisher)"/>
    <param name="initial_engage_state" value="$(var initial_engage_state)"/>
    <param name="perception/enable_detection_failure" value="$(var perception/enable_detection_failure)"/>
    <param name="perception/enable_object_recognition" value="$(var perception/enable_object_recognition)"/>
    <param name="perception/use_base_link_z" value="$(var perception/use_base_link_z)"/>
    <param name="sensing/visible_range" value="$(var sensing/visible_range)"/>
    <param name="scenario_simulation" value="$(var scenario_simulation)"/>
    <param name="rviz" value="$(var rviz)"/>
    <param name="rviz_config" value="$(var rviz_config)"/>
    <param name="rviz_respawn" value="$(var rviz_respawn)"/>
    <param name="vehicle_simulation" value="$(var vehicle_simulation)"/>
  </node>

</launch>